cql3/query_processor.cc

```diff
@@ -59,7 +60,7 @@ static service::query_state query_state_for_internal_call() {
     return {service::client_state::for_internal_calls(), empty_service_permit()};
 }
 
-query_processor::query_processor(service::storage_proxy& proxy, data_dictionary::database db, service::migration_notifier& mn, query_processor::memory_config mcfg, cql_config& cql_cfg, utils::loading_cache_config auth_prep_cache_cfg, wasm::manager& wasm)
+query_processor::query_processor(service::storage_proxy& proxy, data_dictionary::database db, service::migration_notifier& mn, query_processor::memory_config mcfg, cql_config& cql_cfg, utils::loading_cache_config auth_prep_cache_cfg, wasm::manager& wasm, bool maintanance_mode)
         : _migration_subscriber{std::make_unique<migration_subscriber>(this)}
```

Comment:
Why is it enough to disable group0 use in the query_processor only? What about Alternator, thrift and may be some other protocols. What if topology operation is attempted in the recovery mode. Should't the check be in the guard to cover everything?

Reply:
Good point. We didn't plan to enable the alternator etc. But it can be better to disable in all protocols. I will change that to change in the guard.